tests/functional/s/slots_checks.py

```diff
@@ -64,7 +64,7 @@ class SixthBad:  # [single-string-used-for-slots]
     __slots__ = "a"
 
 class SeventhBad:  # [single-string-used-for-slots]
-    __slots__ = ('foo')
+    __slots__ = ('foo')  # [superfluous-parens]
```

Comment:
Same as above, do we make a special case for __slots__ to not fire superlous-parens?

Reply:
I think so because the existing message is more accurate in this scenario.

Reply:
Sorry for not quite following the example. Can you help me understand the case for not emitting a message here?

Reply:
Both `__all__` and `__slots__` are special cases when if a user assigns them to `('foo')`, that evaluates to just string `foo`. 
Both examples currently raise messages `single-string-used-for-slots` or `invalid-all-format`, respectively, because instead of `('foo')`  we expect `('foo', )` (tuple, not str).

However, with this PR, both will now also raise `superfluous-parens`.

So the question to decide is:  are we OK with both examples raising BOTH messages? Or should we do some special casing for __all__ / __slots__ that if we see superfluous parens being raised along with `single-string-used-for-slots` or `invalid-all-format`, we prioritize the latter msg, which is more "accurate"?

If the latter, prioritizing a more specific/ accurate msg over the other, are there other locations where pylint does this already that I can take as an example? I can think of a few ways to do it, but I want to be consistent.

Reply:
Ah, I see the question now. Yes, we should raise both messages. That allows users fine-grained control (especially if they have one message globally disabled, we don't want to rob them of the opportunity to hear about the other message if we just "pick" one.)

Reply:
cool, then code stays as is

Reply:
We can check the slots checker is enabled first, right? The problem is that users will see two messages, one of which is made redundant by the other.

Reply:
That could cause problems with `useless-suppression` etc. IMO it's better to not introduce closer coupling.

Reply:
OK @jacobtylerwalls. Having said all that, I would say 'superfluous-parens' should never be emitted for `__slots__` anyway since slots cannot be a string.

Reply:
I think it [can](https://docs.python.org/3/reference/datamodel.html?highlight=__slots__#object.__slots__) be a string. I can also try to dig up some place we had a conversation about useless-suppression so that I'm sure I'm not misremembering :-D

Reply:
You're right :D I must have remembered it as something that was always a container. Perhaps confusing that we have the `single-string-used-for-slots` message.

But anyway enough talk from me; thanks for the discussion!